### PR TITLE
Expose agent telemetry on system-probe UDS

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Exposes agent telemetry on the system-probe UDS

### Motivation

This will be more efficient and not interfere with USM metrics internally.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/system-probe/api/server.go [hunk 1]
```diff
@@ -45,6 +46,7 @@ func StartServer(cfg *config.Config) error {
 	mux.HandleFunc("/module-restart/{module-name}", restartModuleHandler).Methods("POST")
 
 	mux.Handle("/debug/vars", http.DefaultServeMux)
+	mux.Handle("/telemetry", telemetry.Handler())
```
Comment (val06) [inline cmd/system-probe/api/server.go:49]: what about the `/telemetry` that is initialized [here](https://github.com/DataDog/datadog-agent/blob/main/cmd/system-probe/subcommands/run/command.go#L214) ?

**Review by val06 — CHANGES_REQUESTED** [review]
cred

File: cmd/system-probe/api/server.go (hunk 1 above)
Comment (brycekahle) [reply cmd/system-probe/api/server.go:49]: That is on the default mux, which is exposed only on HTTP.

Comment (val06) [reply cmd/system-probe/api/server.go:49]: talked about that

**Review by val06 — APPROVED** [review]

**Pull request merged**